src/model/model_scaling.jl

```diff
@@ -0,0 +1,261 @@
+    # Else, we return the current proxy variable and coefficient
+    # as the best we can do given the user's coefficient range.
+    # In the future we could try to balance this better with
+    # the RHS scaling
+    else
+        return (proxy_var, new_coeff)
+    end
+end
+
+function prune_coefficients(new_coeff::Real, coeff::Real, multiplier::Real)
+    if new_coeff ≈ 1.0
+        return (1.0, new_coeff / coeff)
+    elseif new_coeff ≈ -1.0
+        return (-1.0, new_coeff / coeff)
+    end
+    return (new_coeff, multiplier)
+end
+
+function calc_rhs_multiplier(con_ref::ConstraintRef, rhs_lb::Real, rhs_ub::Real, coeff_lb::Real, coeff_ub::Real)
+    rhs = normalized_rhs(con_ref)
+    abs_rhs = abs(rhs)
+    if rhs_lb < abs_rhs < rhs_ub
+        return 1.0
+    end
+    coeff_and_rhs = abs.(append!(constraint_object(con_ref).func.terms.vals, rhs))
+    coeff_and_rhs = coeff_and_rhs[coeff_and_rhs .> 0] # Ignore coefficients which equal zero
+    if abs_rhs > rhs_ub
+        return maximum([1.0 / abs_rhs, coeff_lb / coeff_ub / minimum(coeff_and_rhs)])
+    end
+    if abs_rhs < rhs_lb
+        return minimum([1.0 / abs_rhs, coeff_ub / coeff_lb / maximum(coeff_and_rhs)])
+    end
+end
+
+function calc_coeff_multiplier(coeff::Real, coeff_lb::Real, coeff_ub::Real)
+    abs_coeff = abs(coeff)
+    if abs_coeff < coeff_lb
+        return minimum([coeff_ub, 1.0 / abs_coeff]) # We could shift the target value (i.e. 1.0 here)
+    end
+    if abs_coeff > coeff_ub
+        return maximum([coeff_lb, 1.0 / abs_coeff])
+    end
+end
+
+function get_proxy_var(var::VariableRef, multiplier::Real, proxy_var_map::Dict{VariableRef, Vector{Tuple{VariableRef, Float64}}}, proxy_var_ratio_ub::Real)
+    cached_result = existing_proxy_var(var, multiplier, proxy_var_map, proxy_var_ratio_ub)
+    if !isnothing(cached_result)
+        return cached_result
+    end
+    proxy_var = make_proxy_var(var, multiplier)
+    if !haskey(proxy_var_map, var)
+        proxy_var_map[var] = Vector{Tuple{VariableRef, Float64}}()
+    end
+    push!(proxy_var_map[var], (proxy_var, multiplier))
+    return proxy_var, multiplier
+end
+
+function existing_proxy_var(var::VariableRef, multiplier::Real, proxy_var_map::Dict{VariableRef, Vector{Tuple{VariableRef, Float64}}}, proxy_var_ratio_ub::Real)
+    if !haskey(proxy_var_map, var)
+        return nothing
+    end
+    for (proxy_var, cached_multiplier) in proxy_var_map[var]
+        if 1 / proxy_var_ratio_ub < multiplier / cached_multiplier < proxy_var_ratio_ub
+            return proxy_var, cached_multiplier
+        end
+    end
+    return nothing
+end
+
+function make_proxy_var(var::VariableRef, multiplier::Real)
+    model = var.model
+    proxy_var = @variable(model)
+    if has_lower_bound(var)
+        set_lower_bound(proxy_var, lower_bound(var) * multiplier)
+    end
+    if has_upper_bound(var)
+        set_upper_bound(proxy_var, upper_bound(var) * multiplier)
+    end
+    @constraint(model, var == proxy_var * multiplier)
```

Comment:
This may make the coefficients look nicer, but it doesn't actually help.

See https://www.gurobi.com/documentation/current/refman/avoid_hiding_large_coeffic.html

You may also want to take a read of https://jump.dev/JuMP.jl/dev/tutorials/getting_started/tolerances/